### ISSUE TITLE
An unexpected error caused by invalid URLs

### ISSUE THREAD

**Issue by jancurn** [apify/proxy-chain#84]
We've got this in logs:

```
Aug 30 08:31:58 apify-proxy [ERROR] {
  "time":"2020-08-30T08:31:58.623Z",
  "level":"ERROR",
  "msg":"Unexpected error in ProxyChain server",
  "url":"http://i.gj.@/",
  "exception":{
    "name":"Error",
    "message":"The \"trgParsed.hostname\" option is required",
    "stack":"Error: The \"trgParsed.hostname\" option is required\n    at HandlerForward.HandlerBase (/app/node_modules/proxy-chain/build/handler_base.js:51:40)\n    at new HandlerForward (/app/node_modules/proxy-chain/build/handler_forward.js:38:122)\n    at /app/node_modules/proxy-chain/build/server.js:205:31\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)"}}
```

**Comment by jancurn** [comment]
Same for `"url":"http://i.gj.77@/"`

**Comment by jancurn** [comment]
Another one:

```
apify-proxy [ERROR] {"time":"2021-02-14T15:35:36.422Z","level":"ERROR","msg":"Unexpected error in ProxyChain server","url":"http://@/","exception":{"name":"Error","message":"The \"trgParsed.hostname\" option is required","stack":"Error: The \"trgParsed.hostname\" option is required\n    at HandlerForward.HandlerBase (/home/node/node_modules/proxy-chain/build/handler_base.js:51:40)\n    at new HandlerForward (/home/node/node_modules/proxy-chain/build/handler_forward.js:38:122)\n    at /home/node/node_modules/proxy-chain/build/server.js:209:31\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)"}}
```

**Comment by jancurn** [comment]
Should be fixed in 1.0.0
